datalad_metalad/extract.py

```diff
@@ -349,116 +351,82 @@ def custom_result_renderer(res, **kwargs):
             ui.message(json.dumps(context))
 
 
-def do_dataset_extraction(ep: ExtractionArguments):
-
-    if not issubclass(ep.extractor_class, MetadataExtractorBase):
-
-        lgr.debug(
-            "performing legacy dataset level metadata "
-            "extraction for dataset at at %s",
-            ep.source_dataset.path)
-
-        yield from legacy_extract_dataset(ep)
-        return
-
-    if not issubclass(ep.extractor_class, DatasetMetadataExtractor):
-        raise ValueError(
-            "A dataset-level metadata-extraction was attempted (since no "
-            "path argument was given), but the specified extractor "
-            f"({ep.extractor_name}) is not a dataset-level extractor.")
-
-    lgr.debug(
-        "extracting dataset level metadata for dataset at %s",
-        ep.source_dataset.path)
-
-    extractor = ep.extractor_class(
-        ep.source_dataset,
-        ep.source_dataset_version,
-        ep.extraction_parameter)
-
-    yield from perform_dataset_metadata_extraction(ep, extractor)
-
-
-def do_file_extraction(ep: ExtractionArguments):
+def do_extraction(ep: ExtractionArguments, extractor_type: str):
```

Comment:
```suggestion
def do_extraction(ep: ExtractionArguments):
    extractor_type = ep.extractor_type    
```

(add extractor_type to ExtractionArguments)